spec/migrations/20240521100629_add_vault_credential_id_to_options_in_service_templates_spec.rb

```diff
@@ -0,0 +1,43 @@
+require_migration
+
+describe AddVaultCredentialIdToOptionsInServiceTemplates do
+  let(:service_template_stub) { migration_stub(:ServiceTemplate) }
+
+  migration_context :up do
+    it "adds vault_credential_id to provision and retirement options" do
+      service_template = service_template_stub.create!(:options => service_template_up_options)
+      migrate
+      expect(service_template.reload.options[:config_info][:provision][:vault_credential_id]).to eq('provision_credentials')
+      expect(service_template.reload.options[:config_info][:retirement][:vault_credential_id]).to eq('retirement_credentials')
+    end
+  end
+
+  migration_context :down do
+    it "does not clear vault_credential_id if provision or retirement options are not present" do
+      service_template = service_template_stub.create!(:options => service_template_down_options)
+      migrate
+      expect(service_template.reload.options[:config_info][:provision][:vault_credential_id]).to eq(nil)
+      expect(service_template.reload.options[:config_info][:retirement][:vault_credential_id]).to eq(nil)
+    end
+  end
```

Comment:
I think this test is wrong.  In the down the migration will convert vault_credential_id to vault_credentials, so I think this should be:

```suggestion
  migration_context :down do
    it "converts vault_credential_id to vault_credentials in provision and retirement options" do
      service_template = service_template_stub.create!(:options => service_template_down_options)
      migrate
      expect(service_template.reload.options[:config_info][:provision].to_not have_key(:vault_credential_id)
      expect(service_template.reload.options[:config_info][:provision][:vault_credentials]).to eq("provision_credentials")
      expect(service_template.reload.options[:config_info][:retirement]).to_not have_key(:vault_credential_id) 
      expect(service_template.reload.options[:config_info][:retirement][:vault_credentials]).to eq("retirement_credentials")
    end
  end
```

and then below change the options